pennylane/ops/qubit/matrix_ops.py

```diff
@@ -241,6 +241,10 @@ def __init__(
         total_wires = control_wires + wires
         super().__init__(*params, wires=total_wires, do_queue=do_queue)
 
+    @staticmethod
+    def compute_decomposition(*params, wires=None, **hyperparameters):
+        raise DecompositionUndefinedError
```

Comment:
should we add control_wires as a parameter? or save that for later?

Reply:
@puzzleshark hmm the signature should accept that now too, right? Or we could make it more explicit to have it
```python
def compute_decomposition(U, control_wires, u_wires, control_values=None):
```

Reply:
ya I was wondering if we should make it more explicit. good with either way though

Reply:
Got it! Let's save it for later, perhaps we can provide some parts of the decomposition down the line and make the signature more explicit too.